roles/gcp/compute_instance/tasks/create.yml

```diff
@@ -1,63 +1,36 @@
 ---
-- name: Find an image
-  gcp_compute_image_info:
-    project: '{{ image_project }}'
-    auth_kind: '{{ gcp_cred_kind }}'
-    service_account_file: '{{ gcp_cred_file }}'
-    filters:
-    - '{{ compute_image_filters }}'
-  register: result
-
 - set_fact:
-    disk_source_image: '{{ result.resources|last }}'  ## Use the last/newest image
+    node_disk_data: {}
+    node_network_data: {}
 
-- name: Create a disk
-  gcp_compute_disk:
-    name: '{{ disk_name }}'
-    project: '{{ gcp_project }}'
-    auth_kind: '{{ gcp_cred_kind }}'
-    service_account_file: '{{ gcp_cred_file }}'
-    size_gb: '{{ disk_size }}'
-    source_image: '{{ disk_source_image.selfLink }}'
-    zone: '{{ zone }}'
-    state: present
-  register: disk
+## Currently, only one access config, ONE_TO_ONE_NAT, is supported by the Google APIs
+- name: Check if only one access config is provided
+  assert:
+    that:
+      - item.ips | length | int == 1
+  with_items: "{{ nodes }}"
 
-- name: Create a network
-  gcp_compute_network:
-    name: '{{ network_name }}'
-    project: '{{ gcp_project }}'
-    auth_kind: '{{ gcp_cred_kind }}'
-    service_account_file: '{{ gcp_cred_file }}'
-    state: present
-  register: network
+- include_tasks: create_disks.yml
+  with_items: "{{ nodes }}"
 
-- name: Create an address
-  gcp_compute_address:
-    name: '{{ instance_address_name }}'
-    region: '{{ region }}'
-    project: '{{ gcp_project }}'
-    auth_kind: '{{ gcp_cred_kind }}'
-    service_account_file: '{{ gcp_cred_file }}'
-    state: present
-  register: address
+- include_tasks: create_networks.yml
+  with_items: "{{ nodes }}"
 
 - name: Create an instance
   gcp_compute_instance:
-    name: '{{ instance_name }}'
     project: '{{ gcp_project }}'
     auth_kind: '{{ gcp_cred_kind }}'
-    zone: '{{ zone }}'
     service_account_file: '{{ gcp_cred_file }}'
-    machine_type: '{{ machine_type }}'
-    deletion_protection: '{{ deletion_protection }}'
-    disks:
-    - source: '{{ disk }}'
-      auto_delete: 'true'
-      boot: 'true'
+
+    name: '{{ node.name }}'
+    zone: '{{ node.zone }}'
+    machine_type: '{{ node.machine_type }}'
+    deletion_protection: '{{ node.deletion_protection }}'
+    disks: "{{ node_disk_data[node.name] }}"
```

Comment:
it could may be interesting to add a dedicated tasks file to validate this dictionary because if people is overriding the nodes and miss some items this will lead to an unexpected failed

Reply:
Make sense. Need to think about a schema validator.